### PR TITLE
Add anonymous usage reporting for the Opik installer

### PR DESCRIPTION
**Pull request by jverre**
## Details

When `opik install` or `opik upgrade` is called, an anonymous usage report will be submitted to Comet. This can be disabled by setting the environment variable:

```
export OPIK_REPORTING_ENABLED=false
```

It is enabled by default.

### CODE REVIEW

File: deployment/installer/opik_installer/reporting.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    event["event_properties"]["architecture"] = platform.machine()
+    
+    return event
+
+def _create_install_event(success: bool, failure_reason: str = None):
+    event = {
+        "user_id": "Anonymous",
+        "event_type": "opik_installer_install",
+        "event_properties": {
+            "successful": success,
```
Comment (Nimrod007) [inline deployment/installer/opik_installer/reporting.py:20]: I would add OS type and opik version to the events

**Review by Nimrod007 — CHANGES_REQUESTED** [review]

File: deployment/installer/opik_installer/reporting.py (hunk 1 above)
Comment (jverre) [reply deployment/installer/opik_installer/reporting.py:20]: Both are added through the _add_event_metadata function

Comment (jverre) [reply deployment/installer/opik_installer/reporting.py:20]: We have:
    event["event_properties"]["opik_installer_version"] = version.__version__
    event["event_properties"]["os"] = platform.system()
    event["event_properties"]["os_version"] = platform.version()
    event["event_properties"]["architecture"] = platform.machine()

Comment (Nimrod007) [reply deployment/installer/opik_installer/reporting.py:20]: worth getting python version as well ?
```sys.version.split()[0]```

**Review by Nimrod007 — APPROVED** [review]
please done merge lets fix stats server first

**Pull request closed without merging**